src/script/self/SelfRepository.ts

```diff
@@ -208,31 +128,23 @@ export class SelfRepository {
       return localSupportedProtocols;
     }
 
-    return this.updateSelfSupportedProtocols(refreshedSupportedProtocols);
+    await this.updateSelfSupportedProtocols(refreshedSupportedProtocols);
+    return refreshedSupportedProtocols;
   };
 
   /**
    * Will initialise the intervals for checking (and updating if necessary) self supported protocols.
    * Should be called only once on app load.
-   *
-   * @param selfUser - self user
-   * @param teamState - team state
-   * @param userRepository - user repository
    */
   public async initialisePeriodicSelfSupportedProtocolsCheck() {
     // We update supported protocols of self user on initial app load and then in 24 hours intervals
-    const refreshProtocolsTask = async () => {
-      await this.refreshSelfSupportedProtocols();
-      try {
-      } catch (error) {
-        this.logger.error('Failed to update self supported protocols, will retry after 24h. Error: ', error);
-      }
-    };
-    await refreshProtocolsTask();
+    await this.refreshSelfSupportedProtocols();
 
     await this.core.recurringTaskScheduler.registerTask({
       every: TIME_IN_MILLIS.DAY,
-      task: refreshProtocolsTask,
+      task: async () => {
+        await this.refreshSelfSupportedProtocols();
+      },
```

Comment:
```suggestion
      task: () => this.refreshSelfSupportedProtocols()
```

maybe we could change the signature of a `Task` so that it allows `unknown | Promise<unknown>` rather than `void` to avoid those situations. WDYT?

Reply:
Yeah, makes sense, will update it

Reply:
done with https://github.com/wireapp/wire-webapp/pull/15933/commits/37abf0972d48785b6c0d3b9f2ae419d9ed55583c